documentation/projects/proposals/document_all_media_properties/20240313-implementation_plan_document_media_properties_api.md

```diff
@@ -0,0 +1,210 @@
+- The `max_length` data-point is not captured because it is automatically
+  covered under the `db_type` data-point.
+- `is_pk`, `needs_unique`, `allows_null` and `allows_blank` are presented
+  collectively as "constraints" although not technically DB-level constraints.
+```
+
+The information extracted from this introspection will be stored in memory for
+the next stage.
+
+### Merge with additional notes
+
+Additional notes or hardcoded information may be extra information about a field
+such as known inconsistencies or special cases. This information, if documented
+using a consistent format, can be used to augment the information extracted in
+the previous step.
+
+We will document this information in the docstring of the model classes from
+where it can be accessed using `inspect.getdoc`. To ensure that these notes are
+co-located with the fields' code, we will parse docstrings of all ancestors and
+mixins of a model class.
+
+Additional information about fields will be specific inside the model docstring
+as valid YAML under the "Properties" heading. Note the format should be exactly
+as follows with "Properties" being the final subheading.
+
+```python
+class Media(AbstractMedia):
+    """
+    This is a generic media model. In the real world, assume this to be
+    ``Image`` or ``Audio``. It can have its own docstring here.
+
+    Properties
+    ==========
+    title: >-
+        the title of the media; This documentation can span multiple lines and
+        _can contain Markdown_ as it will be injected as-is into the doc page.
+    """
+
+    pass
+```
+
+### Generate documentation page
+
+With all information in-memory about the media properties, the output will be
+written to a page by the management command. This requires some restructuring in
+the pages created for the catalog media properties documentation.
+
+Similar to the catalog documentation, the output document will include the
+`preamble.md` file from `api/api/docs/media_properties/preamble.md`. This will
+be followed by the Markdown generated from the media properties as a table.
+
+### Verify freshness using CI
+
+Using the existing precedent for API recipes being used in the API tests in the
+CI + CD workflow
+([`django-checks`](https://github.com/WordPress/openverse/blob/68ffb8d4a3b4a2336d25a08972285678f446ec90/.github/workflows/ci_cd.yml#L436)),
+we will invoke `gen-media-props` and check for a zero return status.
+
+## Workstreams
+
+The entire plan can be implemented in one stream. A lot of components of the
+plan are shared with the catalog media properties documentation but code reuse
+between the two projects will be limited as they follow different idioms.
+
+The project assumes that the implementation of the catalog documentation project
+will be completed before this project begins, and includes some reorganisation
+of the output of that project. If that is not the case, the reorganisation can
+be incorporated into the ongoing work on that project.
```

Comment:
I see you were waiting for the catalog workstream to be completed. However, I believe the file's final destination should be defined in this IP before starting the code work. It's okay if it's changed later for any (justified) reason.

Reply:
I've added the new hierarchy and the file destination in 416e492.